src/app/register/_components/EmailPasswordBox.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import * as React from "react";
+import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface EmailPasswordBoxProps {
+  setPasswordMatch: (isMatch: boolean) => void;
+  setPasswordValid: (isValid: boolean) => void;
+  setEmailValid: (isValid: boolean) => void;
+  setEmailAlreadyReg: (isValid: boolean) => void;
+}
+
+export default function EmailPasswordBox(props: EmailPasswordBoxProps) {
```

Comment:
instead of doing props.XXX every time.
do something like `const {setPasswordMatch, setPasswordValid, setEmailValid, setEmailAlreadyReg} = props` instead